downstream.go

```diff
@@ -757,6 +763,54 @@ func (dc *downstreamConn) handleMessageUnregistered(ctx context.Context, msg *ir
 		}
 
 		dc.SendMessage(ctx, generateAwayReply(dc.away != nil))
+	case "METADATA":
+		var subcommand string
+		if err := parseMessageParams(msg, nil, &subcommand); err != nil {
+			return err
+		}
+		switch subcommand {
```

Comment:
Seems like this `switch` is identical pre-conn-reg and post-conn-reg - could we dedup these?